src/deprecated.jl

```diff
@@ -1,18 +1,23 @@
-using Base.depwarn
+import Base: @deprecate, depwarn
 
-Base.@deprecate by(d::AbstractDataFrame, cols, s::Vector{Symbol}) aggregate(d, cols, map(eval, s))
-Base.@deprecate by(d::AbstractDataFrame, cols, s::Symbol) aggregate(d, cols, eval(s))
-Base.@deprecate nullable!(colnames::Array{Symbol,1}, df::AbstractDataFrame) nullable!(df, colnames)
-Base.@deprecate nullable!(colnums::Array{Int,1}, df::AbstractDataFrame) nullable!(df, colnums)
+@deprecate by(d::AbstractDataFrame, cols, s::Vector{Symbol}) aggregate(d, cols, map(eval, s))
+@deprecate by(d::AbstractDataFrame, cols, s::Symbol) aggregate(d, cols, eval(s))
+@deprecate nullable!(colnames::Array{Symbol,1}, df::AbstractDataFrame) nullable!(df, colnames)
+@deprecate nullable!(colnums::Array{Int,1}, df::AbstractDataFrame) nullable!(df, colnums)
 import Base: keys, values, insert!
-Base.@deprecate keys(df::AbstractDataFrame) names(df)
-Base.@deprecate values(df::AbstractDataFrame) DataFrames.columns(df)
-Base.@deprecate insert!(df::DataFrame, df2::AbstractDataFrame) merge!(df, df2)
+@deprecate keys(df::AbstractDataFrame) names(df)
+@deprecate values(df::AbstractDataFrame) DataFrames.columns(df)
+@deprecate insert!(df::DataFrame, df2::AbstractDataFrame) merge!(df, df2)
 
 import DataArrays: array, DataArray
-Base.@deprecate array(df::AbstractDataFrame) convert(Array, df)
-Base.@deprecate array(r::DataFrameRow) convert(Array, r)
+@deprecate array(df::AbstractDataFrame) convert(Array, df)
+@deprecate array(r::DataFrameRow) convert(Array, r)
 if VERSION < v"0.4.0-"
-    Base.@deprecate DataArray(df::AbstractDataFrame) convert(DataArray, df)
+    @deprecate DataArray(df::AbstractDataFrame) convert(DataArray, df)
+end
+@deprecate DataArray(df::AbstractDataFrame, T::DataType) convert(DataArray{T}, df)
+
+function read_rda(args...)
+    depwarn("read_rda is deprecated. Use the load function from the RData package.")
+    RData.load(args...)
```

Comment:
Here we should call `FileIO.load()`, because the file format-specific `load()` functions are not meant to be called directly from the user code (and they have special signatures), rather by `FileIO.load()` that delegates loading to specific package based on the current registry of formats. So the more accurate deprecation message would be something like:
```
`read_rda()` is deprecated. R data format support is moved to `RData` package. Use `load()` function.
```

It is also possible to remove circular dependency and replace `RData` with `FileIO` in `REQUIRE`. If `read_rda()` is called before `using RData` (i.e. ".rda" not registered), `FileIO.load()` will give an error about unsupported data type (or `read_rda()` could check whether ".rda" format is registered in `FileIO` and suggest the user to run `using RData`).